perfkitbenchmarker/scripts/object_storage_api_tests.py

```diff
@@ -187,32 +187,55 @@ class SizeDistributionIterator(object):
     dist: dict. The distribution to draw sizes from.
   """
 
-  # IMPLEMENTATION LIMITATION: right now, this object only supports
-  # point distributions, which contain only a single size.
-
   def __init__(self, dist):
-    if len(dist) != 1:
-      raise NotImplementedError(
-          'General distributions are not implemented yet.')
-
-    size, _ = dist.iteritems().next()
-    self.size = size
+    # This class chooses random numbers by using random.random() to
+    # pick a float in [0.0, 100.0), then converting that into a choice
+    # from our distribution. The data structure is two lists, sizes
+    # and cutoffs. If the random number is in the range [cutoffs[i-1],
+    # cutoffs[i]), then the random choice is sizes[i]. Cutoffs has an
+    # implied starting element 0.0, and its last element must be
+    # 100.0.
+    self.sizes = []
+    self.cutoffs = []
+
+    total_percent = 0.0
+    for size, percent in dist.iteritems():
+      total_percent += percent
+      self.sizes.append(size)
+      self.cutoffs.append(total_percent)
+
+    # I considered storing probabilities instead of percents, but I'm
+    # afraid that there is an edge case where the probability
+    # corresponding to a percent doesn't have a terminating binary
+    # representation and then the total percent is less than 1.0 even
+    # though the percentages add up to 100. I can't prove that this
+    # happens, but storing percentages seems safer.
+    assert total_percent == 100.0
```

Comment:
if the user pass in a bad distribution, what would the error message look like? Would it be better if we do an if check, and the print out a friendly error message, and then throw exception? (depends on the answer to the first question, if the error message is obviously useful to the user with the current code, then we are fine)